### ISSUE TITLE
Option for relative positioning to parent

### ISSUE THREAD

**Issue by Nelson-PROIA** [niklashenning/pyqttoast#13]
Great job on the library!

I've been playing around with it, and it's been really useful for notifications. One thing I've noticed is that the positioning seems to be based only on the active screen. Is there a way to make it position relative to its parent element? Maybe I'm missing something, but it would be awesome if you could point me in the right direction!

**Comment by niklashenning** [comment]
Thanks, I appreciate it!

Currently, you can only position the toasts relative to a screen (either the main screen, the screen the parent widget is on, or any other specific fixed screen).
I like the idea of being able to position them relative to the parent widget, though. That's definitely something I could implement.

Just to make sure I understood you correctly, you mean something like in this screenshot where the toasts would be shown in the bottom right corner of the parent widget and not in the bottom right corner of the screen, correct?

![image](https://github.com/user-attachments/assets/b09a5f81-0773-404f-a22a-383c56fdee5d)

**Comment by Nelson-PROIA** [comment]
Yep, that is exactly what I was thinking about!

**Comment by wslpdgg** [comment]
i use Toast.setOffset(x, y) instead , hope your implement 😄

**Comment by niklashenning** [comment]
> i use Toast.setOffset(x, y) instead , hope your implement 😄

I will implement it, don't worry. I'm just really busy for the next two weeks, but after that, it should only take a few days until I release it.
I'll also make it possible for the notifications to move with the parent when it is resized or moved, so the toasts will always stay in the same position relative to the parent if enabled :)

**Comment by niklashenning** [comment]
This feature has now been released in v1.3.0.

After updating to the latest version, you can use the static method `Toast.setPositionRelativeToWidget(widget)`, where the parameter is the widget that the toasts will be relatively positioned to.

Example:
```python
class Window(QMainWindow):
    def __init__(self):
        super().__init__(parent=None)

        self.resize(600, 320)

        # Add button and connect click event
        self.button = QPushButton(self)
        self.button.setText('Show toast')
        self.button.clicked.connect(self.show_toast)

        # Relative to window
        Toast.setPositionRelativeToWidget(self)
        Toast.setOffset(10, 10)

    # Shows a toast notification every time the button is clicked
    def show_toast(self):
        toast = Toast(self)
        toast.setDuration(5000)  # Hide after 5 seconds
        toast.setTitle('Success! Confirmation email sent.')
        toast.setText('Check your email to complete signup.')
        toast.applyPreset(ToastPreset.SUCCESS)  # Apply style preset
        toast.show()
```
